pusher/client.py

```diff
@@ -44,6 +44,8 @@ def __init__(
         self._json_encoder = json_encoder
         self._json_decoder = json_decoder
 
+        self._encryption_master_key = encryption_master_key
```

Comment:
Yes, it is actually only used in the `AuthenticateClient` which originates from the `Client`. I've placed it there since all the other options are set at this level in order to be visible by the other sub-clients. I can place it in the `AuthenticateClient` and then if in the future it is needed somewhere else, it can be re-moved in the main `Client`.

Reply:
Ah, I missed where it was also used in the `AuthenticateClient`. Then I agree, it does make sense to live in the `Client` :)